addons/udes_stock/tests/test_picking_batch.py

```diff
@@ -1705,3 +1705,213 @@ def test04_rereservation_for_same_batch(self):
 
         batch.reserve_pallet('UDES11111')
         batch.reserve_pallet('UDES11111')
+
+
+class TestMultipleOrders(common.BaseUDES):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.pack_4apples_info = [{'product': cls.apple,
+                                  'qty': 4}]
+        cls.pack_4bananas_info = [{'product': cls.banana,
+                                  'qty': 4}]
+
+        cls.picking_type_pick.u_reserve_pallet_per_picking = True
+
+    def setUp(self):
+        super().setUp()
+        self.create_quant(self.apple.id,
+                          self.test_location_01.id,
+                          300,
+                          package_id=self.create_package().id)
+        self.create_quant(self.banana.id,
+                          self.test_location_02.id,
+                          300,
+                          package_id=self.create_package().id)
+
+    def tearDown(self):
+        super().tearDown()
+        self.picking_type_pick.u_max_reservable_pallets = 10
+
+    def create_batch(self, user=None, **kwargs):
+        return super().create_batch(user=user, u_ephemeral=True, **kwargs)
+
+    def test01_associates_pallet_with_initial_batch_picking(self):
+        batch = self.create_batch(user=self.outbound_user)
+        picking1 = self.create_picking(self.picking_type_pick,
+                                       products_info=self.pack_4apples_info,
+                                       confirm=True,
+                                       assign=True,
+                                       batch_id=batch.id)
+        batch.state = 'in_progress'
+        batch.reserve_pallet('UDES11111', picking=picking1)
+
+        self.assertEqual(picking1.u_reserved_pallet, 'UDES11111')
+
+    def test02_associates_pallet_with_batch(self):
```

Comment:
Can we find a better name? We are reserving two pallets to two pickings not to a batch.